src/Components/Aspire.NATS.Net/AspireNatsClientExtensions.cs

```diff
@@ -92,7 +92,26 @@ NatsConnection Factory(IServiceProvider provider)
                 throw new InvalidOperationException($"NATS connection string not found: {connectionName}");
             }
 
-            options = options with { Url = settings.ConnectionString };
+            string redactedConnectionString = settings.ConnectionString;
+            var uri = new Uri(settings.ConnectionString);
+
+            // if authentication via user info is added to NATS.Net this can be removed
+            if (uri.UserInfo.Split(':') is [string user, string password])
+            {
+                options = options with
+                {
+                    AuthOpts = options.AuthOpts with
+                    {
+                        Username = user,
+                        Password = password
+                    }
+                };
+
+                // redact password so it is not visible in logs
+                redactedConnectionString = redactedConnectionString.Replace(password, "***");
```

Comment:
Now after I thought more about this, this can lead to issues if the password is "nats" or another string that is part of the URL. Will change this.